### ISSUE TITLE
Why ignore VCS-based packages accordingly to this message?

### ISSUE THREAD

**Issue by evandrocoan** [wbond/package_control#1233]
This error pops up some times I open Sublime Text. 

I just enabled debug mode, when find the log I post it here.

```
Package Control

Error executing: "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD

Working directory:
D:\SublimeText\Data\Packages\PowerCursors

VCS-based packages can be ignored by changing the "ignore_vcs_packages" setting to true.
Sublime Text will need to be restarted once the setting is changed.
```

**Comment by evandrocoan** [comment]
I recorded the log with debug on:

1. [[LOG] Why ignore VCS-based packages accordingly to this message?](https://gist.github.com/evandrocoan/8b24910c8a9a24066941219d5259d20a)

```
Package Control: Looking for git at: "C:\Program Files\Windows Defender\git.exe", "C:\Program Files (x86)\Skype\Phone\git.exe", "C:\Program Files (x86)\Common Files\Intel\Shared Files\cpp\bin\Intel64\git.exe", "C:\WINDOWS\system32\git.exe", "C:\WINDOWS\git.exe", "C:\WINDOWS\System32\Wbem\git.exe", "C:\WINDOWS\System32\WindowsPowerShell\v1.0\git.exe", "C:\Program Files\Microsoft\Web Platform Installer\git.exe", "C:\Program Files (x86)\Skype\Phone\git.exe", "C:\Program Files (x86)\Windows Kits\8.1\Windows Performance Toolkit\git.exe", "C:\Program Files\Microsoft SQL Server\110\Tools\Binn\git.exe", "C:\Program Files (x86)\QuickTime\QTSystem\git.exe", "C:\Program Files\OpenVPN\bin\git.exe", "C:\Program Files\Microsoft SQL Server\120\Tools\Binn\git.exe", "D:\User\Documents\Lua_5.1\git.exe", "D:\User\Documents\Lua_5.1\clibs\git.exe", "D:\User\Documents\SWI-PrologPortable\App\SWI-Prolog\bin\git.exe", "F:\MATLAB-2015a\runtime\win64\git.exe", "F:\MATLAB-2015a\bin\git.exe", "F:\MATLAB-2015a\polyspace\bin\git.exe", "D:\User\Documents\Calibre2\git.exe", "C:\Program Files\Haskell\bin\git.exe", "D:\User\Documents\Haskell Platform\2014.2.0.0\lib\extralibs\bin\git.exe", "D:\User\Documents\Haskell Platform\2014.2.0.0\bin\git.exe", "D:\User\Documents\Perl64\site\bin\git.exe", "D:\User\Documents\Perl64\bin\git.exe", "C:\ProgramData\Oracle\Java\javapath\git.exe", "C:\Program Files\doxygen\bin\git.exe", "C:\Program Files\Java\jdk1.8.0_65\bin\git.exe", "D:\User\Documents\Vim\vim74\git.exe", "C:\Program Files (x86)\Git\cmd\git.exe", "D:\User\Documents\Cygwin64\bin\git.exe", "D:\User\Documents\MingW\bin\git.exe", "D:\User\Documents\Ruby21-x64\bin\git.exe", "C:\Program Files\PostgreSQL\pg96\bin\git.exe", "D:\User\Documents\latex\LyX 2.2\Perl\bin\git.exe", "F:\Python27\git.exe", "F:\Python27\Scripts\git.exe", "F:\Python27\Library\bin\git.exe", "C:\Program Files (x86)\PostgreSQL\9.6\bin\git.exe", "C:\Users\Professional\AppData\Roaming\cabal\bin\git.exe", "F:\VisualStudio2015\VC\bin\git.exe", "D:\User\DataStore\GamesDataStore\Conter-Strike1.6\Outros\Addons\Multi-Mod\Amx Mod X\CompilerDefault\git.exe", "C:\Users\Professional\AppData\Local\atom\bin\git.exe", "D:\User\Documents\SysinternalsSuite\git.exe", "C:\Users\Professional\.cargo\bin\git.exe", "D:\User\Documents\CMake\bin\git.exe", "D:\User\Documents\latex\texmfs\install\miktex\bin\git.exe", "D:\User\Documents\Octave\bin\git.exe", "C:\Program Files (x86)\SumatraPDF\git.exe", "D:\User\Documents\apache_marven\maven\bin\git.exe", "C:\Program Files (x86)\ActiveState Komodo Edit 10\git.exe", "git.exe", "git.exe", "C:\Program Files\Git\bin\git.exe", "C:\Program Files (x86)\Git\bin\git.exe", "C:\Program Files\TortoiseGit\bin\git.exe", "C:\Program Files\Mercurial\git.exe", "C:\Program Files (x86)\Mercurial\git.exe", "C:\Program Files (x86)\TortoiseHg\git.exe", "C:\Program Files\TortoiseHg\git.exe", "C:\cygwin\bin\git.exe"
Package Control: Found git at "C:\Program Files (x86)\Git\cmd\git.exe"
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\PowerCursors]
error: Package Control

Error executing: "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD

Working directory: D:\SublimeText\Data\Packages\PowerCursors



VCS-based packages can be ignored by changing the "ignore_vcs_packages" setting to true.

Sublime Text will need to be restarted once the setting is changed.
Exception in thread Thread-28:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "D:\SublimeText\Data\Packages\Package Control\package_control\automatic_upgrader.py", line 117, in run
    self.upgrade_packages()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\automatic_upgrader.py", line 257, in upgrade_packages
    ignore_packages=self.auto_upgrade_ignore
  File "D:\SublimeText\Data\Packages\Package Control\package_control\package_installer.py", line 99, in make_package_list
    incoming = upgrader.incoming()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 107, in incoming
    info = self.get_working_copy_info()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 59, in get_working_copy_info
    branch = res.replace('refs/heads/', '')
AttributeError: 'bool' object has no attribute 'replace'
```

**Comment by wbond** [comment]
Did you try executing that command in that directory?

**Comment by evandrocoan** [comment]
Yes, does not output any error or anything:
```
User@User-pc$ git.exe symbolic-ref -q HEAD
User@User-pc$ git.exe symbolic-ref -q HEAD
User@User-pc$
```

**Comment by wbond** [comment]
I believe that means you don't have a remote set up, so PC cant properly manage it

**Comment by evandrocoan** [comment]
I do not understand why I wound need to have a remote setup. What should be remote?

Also, even after I put this on the settings file:
```javascript
	"ignore_vcs_packagess":
	[
		"PowerCursors"
	],
```
That error keeps popping up with the same package. How can it be fixed?

**Comment by wbond** [comment]
The reason a remote needs to be set up is because Package Control is designed to install and update packages. In this case it can't update it, but it appears there is a bug in handling the situation where there is no remote.

I don't know why it is still showing you an error. It sounds like a little debugging is in order.

**Comment by FichteFoll** [comment]
>`ignore_vcs_packagess`

You have an extra `s` there. Not sure if that's the issue

**Comment by evandrocoan** [comment]
Thanks. This was the issue for why package control was not ignoring that package. However, the why package control is getting trouble with that package I am not sure now, but I have a theory.

1. That package `"PowerCursors"` is not installed by package control. It is a fork on my github count. 
1. For all packages which I did not installed by package control, i.e., the packages I forked and cloned, I put them on the list `"auto_upgrade_ignore":[...]`.

But that pacakge `"PowerCursors"` I had forgot to put on the `auto_upgrade_ignore` list, so my guess is the original package on Package control was updated and now package control is trying to update it with the original autor. However the version installed is from my fork.

**Comment by evandrocoan** [comment]
After adding the `"PowerCursors"`  package to the `auto_upgrade_ignore` list, Sublime Text stopped showing that window when sublime text start.

Now I find out Package control cannot install any new packages. Every time I try to that message comes back. Then I disabled that package, and now package control is show the same message, but for another package.

Now I set `"ignore_vcs_packages": true,` instead of only `"ignore_vcs_packages": "PowerCursors",`.
And it worked. The install packages window opened.

**Comment by evandrocoan** [comment]
I recorded a log within the error when I perform the command `Install Package`. Now it is troubling with package `Preferences Editor`. Which is another package I manually installed by `git clone` and had put on the `"auto_upgrade_ignore"` list. 
```
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x32
  Sublime Text Version: 3132
  Package Control Version: 3.2.1
Package Control: Looking for git at: "C:\Program Files\Windows Defender\git.exe", ...
Package Control: Found git at "C:\Program Files (x86)\Git\cmd\git.exe"
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\All Autocomplete]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\All Autocomplete]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\All Autocomplete]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\All Autocomplete]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\All Autocomplete]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\ANSIescape]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\ANSIescape]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\ANSIescape]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\ANSIescape]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\ANSIescape]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\AutoFileName]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\AutoFileName]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\AutoFileName]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\AutoFileName]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\AutoFileName]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\Indent and braces]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\Indent and braces]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\Indent and braces]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\Indent and braces]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\Indent and braces]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\MarkdownLight]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\MarkdownLight]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\MarkdownLight]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\MarkdownLight]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\MarkdownLight]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\Package Control]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.develop.remote [D:\SublimeText\Data\Packages\Package Control]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.develop.merge [D:\SublimeText\Data\Packages\Package Control]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\Package Control]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/develop [D:\SublimeText\Data\Packages\Package Control]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\PackageDev]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.remote [D:\SublimeText\Data\Packages\PackageDev]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" config --get branch.master.merge [D:\SublimeText\Data\Packages\PackageDev]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" fetch origin [D:\SublimeText\Data\Packages\PackageDev]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" log ..origin/master [D:\SublimeText\Data\Packages\PackageDev]
Package Control: Executing "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD [D:\SublimeText\Data\Packages\Preferences Editor]
error: Package Control

Error executing: "C:\Program Files (x86)\Git\cmd\git.exe" symbolic-ref -q HEAD

Working directory: D:\SublimeText\Data\Packages\Preferences Editor



VCS-based packages can be ignored by changing the "ignore_vcs_packages" setting to true.

Sublime Text will need to be restarted once the setting is changed.
Exception in thread Thread-59:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "D:\SublimeText\Data\Packages\Package Control\package_control\commands\install_package_command.py", line 45, in run
    self.package_list = self.make_package_list(['upgrade', 'downgrade', 'reinstall', 'pull', 'none'])
  File "D:\SublimeText\Data\Packages\Package Control\package_control\package_installer.py", line 99, in make_package_list
    incoming = upgrader.incoming()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 107, in incoming
    info = self.get_working_copy_info()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 59, in get_working_copy_info
    branch = res.replace('refs/heads/', '')
AttributeError: 'bool' object has no attribute 'replace'
```

___

## Worked

Now I put all the packages from the `"auto_upgrade_ignore"` list, also on the `"ignore_vcs_packages"` and it worked. Now I can install new packages. This is the log when it was successful:
```
Exception in thread Thread-59:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "D:\SublimeText\Data\Packages\Package Control\package_control\commands\install_package_command.py", line 45, in run
    self.package_list = self.make_package_list(['upgrade', 'downgrade', 'reinstall', 'pull', 'none'])
  File "D:\SublimeText\Data\Packages\Package Control\package_control\package_installer.py", line 99, in make_package_list
    incoming = upgrader.incoming()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 107, in incoming
    info = self.get_working_copy_info()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 59, in get_working_copy_info
    branch = res.replace('refs/heads/', '')
AttributeError: 'bool' object has no attribute 'replace'

reloading settings Packages/User/Package Control.sublime-settings
Package Control: Fetching list of available packages and dependencies
  Platform: windows-x32
  Sublime Text Version: 3132
  Package Control Version: 3.2.1
Package Control: Download Debug
  URL: https://packagecontrol.io/channel_v3.json
  Timeout: 30
  Resolved IP: 50.116.34.243
  Resolved IPv6: 2600:3c02:e000:42::1
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: packagecontrol.io
    issuer: US, GeoTrust Inc., RapidSSL SHA256 CA
    common name: packagecontrol.io
    issue date: Sat, 26 Nov 2016 22:00:00 GMT
    expire date: Wed, 27 Nov 2019 21:59:59 GMT
Package Control: WinINet HTTPS Debug Write
  GET /channel_v3.json HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: packagecontrol.io
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: "592caefe-4707c"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Mon, 29 May 2017 23:30:06 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 200 OK
  Server: nginx
  Date: Mon, 29 May 2017 23:44:32 GMT
  Content-Type: application/json
  Content-Length: 291835
  Last-Modified: Mon, 29 May 2017 23:40:05 GMT
  Connection: keep-alive
  ETag: "592cb155-473fb"
  Content-Encoding: bzip2
  Vary: Accept-Encoding
  Strict-Transport-Security: max-age=15552000
  Accept-Ranges: bytes
Package Control: Caching https://packagecontrol.io/channel_v3.json in D:\SublimeText\Data\Packages\User\Package Control.cache\01524fae79697630d0454ba3fabd9414
Package Control: Download Debug
  URL: https://api.github.com/repos/math2001/FileManager
  Timeout: 30
  Resolved IP: 192.30.253.116
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug General
  Server SSL Certificate:
    subject: US, California, San Francisco, "GitHub, Inc.", *.github.com
    issuer: US, DigiCert Inc, www.digicert.com, DigiCert SHA2 High Assurance Server CA
    common name: *.github.com
    issue date: Tue, 17 Jan 2017 22:00:00 GMT
    expire date: Fri, 17 Apr 2020 09:00:00 GMT
Package Control: WinINet HTTPS Debug Write
  GET /repos/math2001/FileManager HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: api.github.com
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: W/"e4538ec7bf907911b837a6e287bd19c7"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Wed, 17 May 2017 16:39:59 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: GitHub.com
  Date: Mon, 29 May 2017 23:44:37 GMT
  Status: 304 Not Modified
  X-RateLimit-Limit: 60
  X-RateLimit-Remaining: 59
  X-RateLimit-Reset: 1496104489
  Cache-Control: public, max-age=60, s-maxage=60
  Vary: Accept
  ETag: "e4538ec7bf907911b837a6e287bd19c7"
  Last-Modified: Wed, 17 May 2017 16:39:59 GMT
  Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
  Access-Control-Allow-Origin: *
  Content-Security-Policy: default-src 'none'
  Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
  X-Content-Type-Options: nosniff
  X-Frame-Options: deny
  X-XSS-Protection: 1; mode=block
  Vary: Accept-Encoding
  X-Served-By: eef8b8685a106934dcbb4b7c59fba0bf
  X-GitHub-Request-Id: 0E96:29AF:3DF9718:4A32FB9:592CB264
Package Control: Using cached content for https://api.github.com/repos/math2001/FileManager from D:\SublimeText\Data\Packages\User\Package Control.cache\1e003f15473e4cc9db3654bc384207d1
Package Control: Download Debug
  URL: https://api.github.com/repos/math2001/FileManager/readme?ref=master
  Timeout: 30
  Resolved IP: 192.30.253.116
Package Control: WinINet HTTPS Debug General
  Re-using connection to api.github.com on port 443 for request #1
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug Write
  GET /repos/math2001/FileManager/readme?ref=master HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: api.github.com
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: W/"0f1393442778a9d87edb581c1607aa71"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Wed, 19 Apr 2017 01:22:12 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: GitHub.com
  Date: Mon, 29 May 2017 23:44:37 GMT
  Status: 304 Not Modified
  X-RateLimit-Limit: 60
  X-RateLimit-Remaining: 59
  X-RateLimit-Reset: 1496104489
  Cache-Control: public, max-age=60, s-maxage=60
  Vary: Accept
  ETag: "0f1393442778a9d87edb581c1607aa71"
  Last-Modified: Wed, 19 Apr 2017 01:22:12 GMT
  Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
  Access-Control-Allow-Origin: *
  Content-Security-Policy: default-src 'none'
  Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
  X-Content-Type-Options: nosniff
  X-Frame-Options: deny
  X-XSS-Protection: 1; mode=block
  Vary: Accept-Encoding
  X-Served-By: 49aa99f015c25437a7443c4d3a58cd17
  X-GitHub-Request-Id: 0E96:29AF:3DF9741:4A32FE8:592CB265
Package Control: Using cached content for https://api.github.com/repos/math2001/FileManager/readme?ref=master from D:\SublimeText\Data\Packages\User\Package Control.cache\8c6e0447b88b7bcf6c2eeef8243ae75d
Package Control: Download Debug
  URL: https://api.github.com/repos/math2001/FileManager/commits?per_page=1&sha=master
  Timeout: 30
  Resolved IP: 192.30.253.116
Package Control: WinINet HTTPS Debug General
  Re-using connection to api.github.com on port 443 for request #2
Package Control: WinINet Debug Proxy
  proxy: 
  proxy bypass: 
  proxy username: 
  proxy password: 
Package Control: WinINet HTTPS Debug Write
  GET /repos/math2001/FileManager/commits?per_page=1&sha=master HTTP/1.1
  User-Agent: Package Control v3.2.1
  Host: api.github.com
  Connection: Keep-Alive
  Cache-Control: no-cache
  If-None-Match: W/"705feb1d8093f0755d97f36de4cdc540"
  Accept-Encoding: bzip2,gzip,deflate
  If-Modified-Since: Wed, 19 Apr 2017 01:22:12 GMT
Package Control: WinINet HTTPS Debug Read
  HTTP/1.1 304 Not Modified
  Server: GitHub.com
  Date: Mon, 29 May 2017 23:44:37 GMT
  Status: 304 Not Modified
  X-RateLimit-Limit: 60
  X-RateLimit-Remaining: 59
  X-RateLimit-Reset: 1496104489
  Cache-Control: public, max-age=60, s-maxage=60
  Vary: Accept
  ETag: "705feb1d8093f0755d97f36de4cdc540"
  Last-Modified: Wed, 19 Apr 2017 01:22:12 GMT
  Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
  Access-Control-Allow-Origin: *
  Content-Security-Policy: default-src 'none'
  Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
  X-Content-Type-Options: nosniff
  X-Frame-Options: deny
  X-XSS-Protection: 1; mode=block
  Vary: Accept-Encoding
  X-Served-By: 62cdcc2d03a2f173f1c58590d1a90077
  X-GitHub-Request-Id: 0E96:29AF:3DF9762:4A33014:592CB265
Package Control: Using cached content for https://api.github.com/repos/math2001/FileManager/commits?per_page=1&sha=master from D:\SublimeText\Data\Packages\User\Package Control.cache\ca1b03bfe1af3170fc1b82e9ca754bfc
Package Control: WinINet HTTPS Debug General
  Closing connection to api.github.com on port 443 after 3 requests
Package Control: WinINet HTTPS Debug General
  Closing connection to packagecontrol.io on port 443 after 1 request
```

___ 

# New error

There was this exceptions write before the settings reload:
```
Exception in thread Thread-59:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "D:\SublimeText\Data\Packages\Package Control\package_control\commands\install_package_command.py", line 45, in run
    self.package_list = self.make_package_list(['upgrade', 'downgrade', 'reinstall', 'pull', 'none'])
  File "D:\SublimeText\Data\Packages\Package Control\package_control\package_installer.py", line 99, in make_package_list
    incoming = upgrader.incoming()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 107, in incoming
    info = self.get_working_copy_info()
  File "D:\SublimeText\Data\Packages\Package Control\package_control\upgraders\git_upgrader.py", line 59, in get_working_copy_info
    branch = res.replace('refs/heads/', '')
AttributeError: 'bool' object has no attribute 'replace'
```

**Comment by evandrocoan** [comment]
I find out what causes this error. I had the package `All Autocomplete` triggering it. Then I gone to the folder `Packages/All Autocomplete` and noticed it is a git repo synlinked.

It is because the `All Autocomplete` is a submodule, therefore its gits files are in:

1. `gitdir: ../.git/modules/All Autocomplete` 

Now you can reproduce it, but you need:

1. To make your Packages folder a git repository, and add the `All Autocomplete` as a submodule.
1. Delete it, and install clone your `Package`  folder repo with `git clone --recursive`

This will create the `All Autocomplete` as a git submodule and store its files on the parent git folder at:

1. `gitdir: ../.git/modules/All Autocomplete`

**Comment by evandrocoan** [comment]
I just figured out the problem. When I do `git clone --recursive` the submodules are not checkout on their branches, therefore the command ` git.exe symbolic-ref -q HEAD` returns empty.

The solution is just go to each submodule and do `git checkout master` or the main default branch.

**Comment by wbond** [comment]
This was fixed by 790bcd6
